### ISSUE TITLE
Additional country-specific validations

### ISSUE THREAD

**Issue by globalcitizen** [globalcitizen/php-iban#40]
Addition of special case national-level validations, eg. Estonia's record states that the first digit of the bank code cannot be 0, and Slovenia's five digit bank ID always begins with 91. At least one version of the German BLZ (bank IDs) database is available over here: https://github.com/Bigfoot0485/IBAN

**Comment by globalcitizen** [comment]
More generally, we could consider building a library of national-level bank or payment institution identifier codes. This would be rather large and an optional extension and would be in competition with SWIFTRef... see https://swiftrefdata.com/ and https://www.swift.com/our-solutions/interfaces-and-integration/swift-apis/swiftref-api and http://www.slideshare.net/SWIFTcommunity/bbf2015swift-refwork-sessionfinal2jul2015

**Comment by globalcitizen** [comment]
Issue #46 discusses the location and fetching of Spanish BBAN-incorporated checksum digits.

**Comment by globalcitizen** [comment]
http://www.betaalvereniging.nl/en/european-transactions/sepa-documentation-for-the-netherlands-2/bic-from-iban/ is an apparently always up to date list of all Dutch bank IDs, for example `NL91ABNA0417164300` would be `ABNA` which is `ABN AMRO BANK`.

**Comment by globalcitizen** [comment]
http://www.pangaliit.ee/Estonian_Payment_Service_Provider_codes_and_BICs.xlsx is an apparently always up to date list of all Estonian bank IDs, for example `EE382200221020145685` would be `22` which is `Swedbank AS`.

**Comment by globalcitizen** [comment]
I've started this rather maintenance-headache-inducing project in an exploratory fashion. Currently thinking the fields would be:
* __IBAN country__: The country in which the bank's identifier belongs.
* __Bank identifier__: What is present in the appropriate portion of the BBAN for the country in question.
* __Bank title__: In officious English, eg. 'Hong Kong Shanghai Banking Corporation Netherlands Limited'
* __Bank SWIFT BIC__: Where known.
* __Bank short title__: Something like 'HSBC' rather than 'Hong Kong Shanghai Banking Corporation Netherlands Limited'
* __Bank URL__: With prefixed http:// or https:// removed. In most cases the domain name will suffice.

**Comment by globalcitizen** [comment]
[List of banks of Andorra](https://www.inaf.ad/index.php?option=com_multicategories&view=category&id=12%3Aentitats-bancaries&Itemid=76&lang=en).

**Comment by globalcitizen** [comment]
[List of banks and SWIFT codes in Slovenia](http://www.kreatik.si/bic_slovenia.php).

**Comment by globalcitizen** [comment]
Albania: [1](https://plus.al/en/business-how-can-we-help-bill-payment), [2](http://www.shekulli.com.al/p.php?id=213741), [3](http://www.gazetadita.al/permbytjet-ne-jug-llogarite-bankare-ku-mund-te-ndihmoni-banoret-e-prekur/). [4](https://www.tatime.gov.al/sq-al/Media/Lajmet/Pages/Ne-ndihme-te-familjeve-ne-nevoje-nga-permbytjet.aspx).

**Comment by globalcitizen** [comment]
https://snel.it/svn/misc/trunk/au3/BICs-2014-03-07.php seems to be a conversion table from IBAN bankid to SWIFT BIC... rather short though.

**Comment by AlexKratky** [comment]
@globalcitizen 

IBAN validation for the Czech Republic is not 100% correct, there is a rule "a second part containing at least 2 and at most 10 digits, at least two of which may not be zero, with the initial zeros having no significance.". The second part is the account part, so for example CZ0401000000000000000000 should be an invalid IBAN. I have tried to create a regex that will fulfill this rule, but I am not able to create a negative lookahead that will match one non-zero digit at any position. The non-all zeros regex is simple (?![0]{10})(\d{10}), but for one non-zero character not, the only solution that I figured out is to write all scenarios (where the one digit could be) for example (?![0]{10})(?![0]{1}[1-9]{1}[0]{8})...(?![0]{8}[1-9]{1}[0]{1})(\d{10}) which is terrible regex.

Specification:
https://www.cnb.cz/export/sites/cnb/en/legislation/.galleries/decrees/decree_169_2011.pdf

**Comment by Triplkrypl** [comment]
> at least two of which may not be zero, with the initial zeros having no significance.

Yes our CZ bank account number is pain. I also tried create fix, but with one regex, i was out of idea.

**Comment by Triplkrypl** [comment]
Here are some new examples: #121 with some differences from previous ones.
